app/src/main/java/com/example/itbook/ui/search/SearchPresenter.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.itbook.ui.search
+
+import com.example.itbook.data.model.Book
+import com.example.itbook.data.repository.BooksRepository
+import com.example.itbook.data.source.remote.API.APIQuery
+import com.example.itbook.utils.OnDataLoadCallBack
+
+class SearchPresenter(
+    private val view: SearchContract.View,
+    private val repository: BooksRepository,
+    private val query: String
+) : SearchContract.Presenter {
+
+    private val books = mutableListOf<Book>()
+    private var currentPage = 1
+    private var isStop = false
+
+    override fun getSearchBook(query: String) {
+        view.showLoading(true)
+        repository.getRemoteBooks(
+            APIQuery.queryBooks(query, currentPage),
+            object : OnDataLoadCallBack<List<Book>> {
+                override fun onSuccess(data: List<Book>) {
+                    books.addAll(data)
+                    view.showSearchResult(books)
+                    view.showLoading(false)
+
+                    if (data.isNotEmpty() && !isStop) {
+                        currentPage++
+                        repository.getRemoteBooks(APIQuery.queryBooks(query, currentPage), this)
+                    } else if (isStop) view.resetResult()
```

Comment:
```suggestion
                    } else if {
                     (isStop) view.resetResult()
``` nên xuống dòng cho vào ngoặc vì trên  if mở ngoặc r